src/game/components/GameOneVersusOne/GamePlacingShipsView.tsx

```diff
@@ -60,79 +64,77 @@ const GamePlacingShipsView = (): JSX.Element => {
     };
   }, []);
 
-  const gamePlayerIsReady = useMemo(() => !!gamePlayer?.shipsPlacementIsValid, [gamePlayer]);
-  const gameRivalPlayerIsReady = useMemo(
-    () => !!gameRivalPlayer?.shipsPlacementIsValid,
-    [gameRivalPlayer],
-  );
-  const allPlayersAreReady = useMemo(
-    () => gamePlayerIsReady && gameRivalPlayerIsReady,
-    [gameRivalPlayerIsReady, gamePlayerIsReady],
-  );
-
   useEffect(() => {
     if (allPlayersAreReady && counterTime > 0) {
       intervalId = setInterval(() => setCounterTime(counterTime - 1), 1000);
     } else {
       clearInterval(intervalId);
     }
```

Comment:
Use early return principle 
```suggestion
    if (allPlayersAreReady && counterTime > 0) {
      intervalId = setInterval(() => setCounterTime(counterTime - 1), 1000);
      return;
    } 
    
      clearInterval(intervalId);
```

Reply:
:heavy_check_mark: